code/controllers/subsystems/nightshift.dm

```diff
@@ -0,0 +1,62 @@
+SUBSYSTEM_DEF(nightshift)
+	name = "Night Shift"
+	init_order = INIT_ORDER_NIGHTSHIFT
+	priority = FIRE_PRIORITY_NIGHTSHIFT
+	wait = 60 SECONDS
+	flags = SS_NO_TICK_CHECK
+
+	var/nightshift_active = FALSE
+	var/nightshift_first_check = 30 SECONDS
+
+	var/high_security_mode = FALSE
+
+/datum/controller/subsystem/nightshift/Initialize()
+	if(!config.enable_night_shifts)
+		can_fire = FALSE
+	/*
+	if(config.randomize_shift_time)
+		GLOB.gametime_offset = rand(0, 23) HOURS
+	*/
+	return ..()
+
+/datum/controller/subsystem/nightshift/fire(resumed = FALSE)
+	if(round_duration_in_ticks < nightshift_first_check)
+		return
+	check_nightshift()
+
+/datum/controller/subsystem/nightshift/proc/announce(message)
+	var/announce_z
+	if(using_map.station_levels.len)
+		announce_z = pick(using_map.station_levels)
+	priority_announcement.Announce(message, new_title = "Automated Lighting System Announcement", new_sound = 'sound/misc/notice2.ogg', zlevel = announce_z)
+
+/datum/controller/subsystem/nightshift/proc/check_nightshift(check_canfire=FALSE) //This is called from elsewhere, like setting the alert levels
+	if(check_canfire && !can_fire)
+		return
+	var/emergency = security_level > SEC_LEVEL_GREEN
+	var/announcing = TRUE
+	var/night_time = using_map.get_nightshift()
+	if(high_security_mode != emergency)
+		high_security_mode = emergency
+		if(night_time)
+			announcing = FALSE
+			if(!emergency)
+				announce("Restoring night lighting configuration to normal operation.")
+			else
+				announce("Disabling night lighting: Station is in a state of emergency.")
+	if(emergency)
+		night_time = FALSE
+	if(nightshift_active != night_time)
+		update_nightshift(night_time, announcing)
+
+/datum/controller/subsystem/nightshift/proc/update_nightshift(active, announce = TRUE)
+	nightshift_active = active
+	if(announce)
+		if(active)
+			announce("Good evening, crew. To reduce power consumption and stimulate the circadian rhythms of some species, all of the lights aboard the station have been dimmed for the night.")
+		else
+			announce("Good morning, crew. As it is now day time, all of the lights aboard the station have been restored to their former brightness.")
```

Comment:
`standard`, rather than `former`? It's a lot more likely for a round to pass from night to day by virtue of rng picking a predawn time than it is for a round to last long enough that both transitions could be seen

Reply:
Depends on your station/planet. We have 6 hour shifts on a planet with 3 hour days, so it transitions every 1.5 hours. up to 4 times.

The announcement is a copypaste from Paradise either way. If you want me to change it, I can.